python4j/python4j-core/src/main/java/org/eclipse/python4j/Python.java

```diff
@@ -0,0 +1,436 @@
+        }
+
+
+    }
+
+    public static PythonObject str() {
+        PythonGIL.assertThreadSafe();
+        try{
+            return PythonTypes.STR.toPython("");
+        }catch (Exception e){
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static PythonObject strType() {
+        return attr("str");
+    }
+
+    public static PythonObject float_(PythonObject pythonObject)throws PythonException{
+        return PythonTypes.FLOAT.toPython(PythonTypes.FLOAT.toJava(pythonObject));
+    }
+
+    public static PythonObject float_() {
+        try{
+            return PythonTypes.FLOAT.toPython(0d);
+        }catch (Exception e){
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    public static PythonObject floatType() {
+        return attr("float");
+    }
+
+    public static PythonObject bool(PythonObject pythonObject){
+        try{
+            return PythonTypes.BOOL.toPython(PythonTypes.BOOL.toJava(pythonObject));
+        }catch (PythonException pe){
+            throw new RuntimeException(pe);
+        }
+
+    }
+
+    public static PythonObject bool() {
+        try{
+            return PythonTypes.BOOL.toPython(false);
+        }catch (Exception e){
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    public static PythonObject boolType() {
+        return attr("bool");
+    }
+
+    public static PythonObject int_(PythonObject pythonObject) throws PythonException{
+        return PythonTypes.INT.toPython(PythonTypes.INT.toJava(pythonObject));
+    }
+
+    public static PythonObject int_() {
+        try{
+            return PythonTypes.INT.toPython(0L);
+        }catch (Exception e){
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    public static PythonObject intType() {
+        return attr("int");
+    }
+
+    public static PythonObject list(PythonObject pythonObject) throws PythonException{
+        PythonGIL.assertThreadSafe();
+        try(PythonGC _ = PythonGC.watch()){
+            PythonObject listF = attr("list");
+            PythonObject ret = listF.call(pythonObject);
+            if (ret.isNone()){
+                throw new PythonException("Object is not iterable: " + pythonObject.toString());
+            }
+            return ret;
+        }
+    }
+
+    public static PythonObject list() {
+        try{
+            return PythonTypes.LIST.toPython(Collections.emptyList());
+        } catch(PythonException pe){
+            throw new RuntimeException(pe);
+        }
+    }
+
+    public static PythonObject listType() {
+        return attr("list");
+    }
+
+    public static PythonObject dict(PythonObject pythonObject) throws PythonException{
+        PythonObject dictF = attr("dict");
+        PythonObject ret = dictF.call(pythonObject);
+        if (ret.isNone()){
+            throw new PythonException("Cannot build dict from object: " + pythonObject.toString());
+        }
+        dictF.del();
+        return ret;
+    }
```

Comment:
Do we need a `PythonGIL.assertThreadSafe();` here?
I note we have it on `list(PythonObject pythonObject)` but not here... if it's intentional, what's the reasoning?

Also not sure if we need it on things like `int_()` etc (lots of methods here that don't have it.

Reply:
The `attr` function does the thread safe check. So if the first line of a function is an `attr` call, then that function doesn't need the check.